### PR TITLE
Chore/improve code coverage

### LINKED ISSUE #743: Code Coverage Improvements
We should take another pass at our code coverage and see where it may be worth improving it.

https://app.codecov.io/gh/LibertyDSNP/frequency

- [x] RPC tests
- [x] Remove `common/primitives/src/benchmarks.rs` from coverage
- [x] Adding simple tests?

### PR DESCRIPTION
**Pull request by wilwade**
# Goal
The goal of this PR is to improve code coverage

Closes #743

# Discussion
- Added tests
- Removed `default` need for enums
- Removed unused extra avro methods

# Checklist
- [x] Tests added

image

### CODE REVIEW

File: common/helpers/src/avro.rs [hunk 1]
```diff
@@ -34,35 +33,6 @@ pub fn fingerprint_raw_schema(raw_schema: &str) -> Result<(Schema, Vec<u8>), Avr
 	Ok((schema_result, schema_canonical_form.as_bytes().to_vec()))
 }
 
-/// Function to convert a list of raw schema into serialized Avro schema.
```
Comment (wilwade) [inline common/helpers/src/avro.rs:37]: These functions were unused, so removed for now.

File: common/helpers/src/avro_tests.rs [hunk 2]
```diff
@@ -418,32 +418,3 @@ fn test_json_serialized_bad_avro_schema() {
 	let validation_res = avro::validate_raw_avro_schema(&serialized_bytes);
 	assert!(validation_res.is_err());
 }
-
-#[test]
```
Comment (wilwade) [inline common/helpers/src/avro_tests.rs:422]: Removing test of unused functions

File: common/primitives/src/msa.rs [hunk 3]
```diff
@@ -210,3 +210,36 @@ pub struct KeyInfoResponse<AccountId> {
 	/// The MSA associated with the `key`
 	pub msa_id: MessageSourceId,
 }
+
+#[cfg(test)]
```
Comment (wilwade) [inline common/primitives/src/msa.rs:214]: Simple tests of the scale decoding. Worthwhile? Perhaps not, but not bad examples of how to test it.

File: common/primitives/src/utils.rs [hunk 4]
```diff
@@ -133,13 +140,21 @@ mod tests {
...
 		assert!(result.is_ok());
 		assert_eq!("{}", result.unwrap());
 	}
 
+	#[test]
+	fn as_hex_option_can_serialize_nothing_as_null() {
+		let test_data = TestAsHexOptionNull { data: None };
+		let result = serde_json::to_string(&test_data);
+		assert!(result.is_ok());
+		assert_eq!("{\"data\":null}", result.unwrap());
```
Comment (wilwade) [inline common/primitives/src/utils.rs:155]: Simple test for when you use as_hex_option, but don't want to skip it (aka null is ok.)

File: pallets/messages/src/rpc/Cargo.toml [hunk 5]
```diff
@@ -36,4 +36,5 @@ std = [
 	'frame-support/std',
 	"pallet-messages-runtime-api/std",
 	'common-primitives/std',
+	"common-helpers/std",
```
Comment (wilwade) [inline pallets/messages/src/rpc/Cargo.toml:39]: Needed to run tests independently for rpcs

File: pallets/msa/src/rpc/src/tests/rpc_mock.rs [hunk 6]
```diff
@@ -0,0 +1,64 @@
+use sp_api::{ApiRef, ProvideRuntimeApi};
```
Comment (wilwade) [inline pallets/msa/src/rpc/src/tests/rpc_mock.rs:1]: Note this is a duplicate of `pallets/messages/src/rpc/src/tests/rpc_mock.rs`, but didn't want to try to extract it separately.

File: pallets/schemas/src/rpc/src/tests/rpc_mock.rs [hunk 7]
```diff
@@ -0,0 +1,64 @@
+use sp_api::{ApiRef, ProvideRuntimeApi};
```
Comment (wilwade) [inline pallets/schemas/src/rpc/src/tests/rpc_mock.rs:1]: Note this is a duplicate of `pallets/messages/src/rpc/src/tests/rpc_mock.rs`, but didn't want to try to extract it separately.

File: pallets/schemas/src/types.rs [hunk 8]
```diff
@@ -5,7 +5,7 @@ use scale_info::TypeInfo;
 
 use codec::{Decode, Encode, MaxEncodedLen};
 
-#[derive(Default, Clone, Encode, Decode, PartialEq, Debug, TypeInfo, Eq, MaxEncodedLen)]
+#[derive(Clone, Encode, Decode, PartialEq, Debug, TypeInfo, Eq, MaxEncodedLen)]
```
Comment (wilwade) [inline pallets/schemas/src/types.rs:8]: Unneeded Default.

File: pallets/msa/src/rpc/src/tests/mod.rs [hunk 9]
```diff
@@ -0,0 +1,178 @@
...
+const PROVIDER_WITH_DELEGATE_A: ProviderId = ProviderId(1);
+const DELEGATE_A: DelegatorId = DelegatorId(2);
+const DELEGATE_B: DelegatorId = DelegatorId(3);
+const PROVIDER_WITH_DELEGATE_A_AND_B: ProviderId = ProviderId(4);
+const SCHEMA_FOR_A: u16 = 1;
+const SCHEMA_FOR_A_AND_B: u16 = 2;
+const SCHEMA_FOR_B: u16 = 3;
+const NOT_EXIST_MSA: u64 = 100;
+
+sp_api::mock_impl_runtime_apis! {
```
Comment (wilwade) [inline pallets/msa/src/rpc/src/tests/mod.rs:20]: This is the runtime mock. So we are only testing the code in the rpc side of things.

File: pallets/schemas/src/rpc/src/tests/mod.rs [hunk 10]
```diff
@@ -0,0 +1,104 @@
...
+use super::*;
+use rpc_mock::*;
+
+use pallet_schemas_runtime_api::SchemasRuntimeApi;
+use std::sync::Arc;
+use substrate_test_runtime_client::runtime::Block;
+
+const SUCCESSFUL_SCHEMA_ID: u16 = 1;
+
+sp_api::mock_impl_runtime_apis! {
```
Comment (wilwade) [inline pallets/schemas/src/rpc/src/tests/mod.rs:12]: This is the runtime mock. So we are only testing the code in the rpc side of things.

**Review by saraswatpuneet — APPROVED** [review]
:1st_place_medal:

**Pull request merged**